Jenkinsfile

```diff
@@ -1,150 +1,100 @@
 #!/usr/bin/env groovy
 
-def dockerRegistry = "329802642264.dkr.ecr.eu-west-1.amazonaws.com"
-def nodeImageVersion = "10.16.0"
-def nodeImage = "${dockerRegistry}/bbc-news/node-10-lts:${nodeImageVersion}"
+def nodeImage = "329802642264.dkr.ecr.eu-west-1.amazonaws.com/bbc-news/node-10-lts:10.16.0"
 
-def nodeName
-def slackChannel = "#si_repo-psammead"
-def stageName = "Unknown"
-def gitCommitAuthor = "Unknown"
-def gitCommitMessage = "Unknown"
+// def slackChannel = "#si_repo-psammead"
+def slackChannel = "#si_repo-infra-sandbox"
 
-def getCommitInfo = {
-  gitCommitAuthor = sh(returnStdout: true, script: "git --no-pager show -s --format='%an' ${GIT_COMMIT}").trim()
-  gitCommitMessage = sh(returnStdout: true, script: "git log -1 --pretty=%B").trim()
-}
-
-def notifySlack(colour, buildStatus, gitCommitAuthor, stageName, gitCommitMessage, slackChannel) {
+def notifySlack(colour, buildStatus, gitCommitAuthor, info, gitCommit, gitCommitMessage, slackChannel) {
   // call the global slackSend method in Jenkins
-  slackSend(color: colour, message: "*${buildStatus}* on ${GIT_BRANCH} [build ${BUILD_DISPLAY_NAME}] \n*Author:* ${gitCommitAuthor} \n*Stage:* ${stageName} \n*Commit Hash* \n${GIT_COMMIT} \n*Commit Message* \n${gitCommitMessage}", channel: slackChannel)
+  slackSend(color: colour, message: "*${buildStatus}* on ${env.BRANCH_NAME} [build ${BUILD_DISPLAY_NAME}] \n*Author:* ${gitCommitAuthor} \n*Info:* ${info} \n*Commit Hash* \n${gitCommit} \n*Commit Message* \n${gitCommitMessage}", channel: slackChannel)
 }
 
-pipeline {
-  agent any
-  options {
-    timeout(time: 60, unit: 'MINUTES')
-    timestamps ()
-  }
-  environment {
-    CI = true
-    CC_TEST_REPORTER_ID = '06c1254d7c2ff48f763492791337193c8345ca8740c34263d68adcc449aff732'
-  }
-  stages {
-    stage ('Run application tests') {
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
-        }
-      }
-      steps {
-        sh 'ls' // quick debug step
-        sh 'rm -rf ./app'
-        script {
-          if (GIT_BRANCH == 'latest') {
-            getCommitInfo()
+node {
+  timeout(time: 30, unit: 'MINUTES') {
+    withEnv([
+      'CI=true',
+      'CC_TEST_REPORTER_ID=06c1254d7c2ff48f763492791337193c8345ca8740c34263d68adcc449aff732'
+    ]) {
+      // git checkout
+      checkout scm
+
+      // get git commit info for notifications
+      gitCommitHash = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
+      gitCommitAuthor = sh(returnStdout: true, script: "git --no-pager show -s --format='%an' ${gitCommitHash}").trim()
+      gitCommitMessage = sh(returnStdout: true, script: "git log -1 --pretty=%B").trim()
+
+      docker.image("${nodeImage}").inside {
+        try {
+          stage ('Setup & Install') {
+
+            sh 'make setup-git'
+            sh 'npm run ci:packages'
+            sh 'npm run build:storybook'
```

Comment:
Wondering about the mix-and-match of `npm` scripts and `Makefile` targets that are used here and in the following stage(s), is there an ongoing reason to need both or is there an opportunity to get more consistency/simplicity through this refactor?

Reply:
Whats your preference here? Im happy with using just the makefile.

Reply:
If they are intended to be run only on CI inside a known Docker image personally I'd go for Makefile targets as they are more flexible than `npm` scripts but I don't have a very strong preference, just think it's not currently clear why we are using a combination of both.

Reply:
👍 have moved to using the makefile for all pipeline steps